app/models/manageiq/providers/openstack/network_manager/network_router.rb

```diff
@@ -2,10 +2,26 @@ class ManageIQ::Providers::Openstack::NetworkManager::NetworkRouter < ::NetworkR
   include ProviderObjectMixin
   include AsyncDeleteMixin
 
-  supports :create_network_router
-  supports :delete_network_router
-  supports :update_network_router
   supports :add_interface
+
+  supports :create
+
+  supports :delete do
+    if ext_management_system.nil?
+      unsupported_reason_add(:delete_network_router, _("The Network Router is not connected to an active %{table}") % {
```

Comment:
the reason should match the one in the `supports` block.
` unsupported_reason_add(:delete ...`